sycl/source/detail/sampler_impl.cpp

```diff
@@ -36,18 +36,33 @@ sampler_impl::sampler_impl(cl_sampler clSampler, const context &syclContext)
 
 sampler_impl::~sampler_impl() {
   if (m_ReleaseSampler)
-    CHECK_OCL_CODE(clReleaseSampler(m_Sampler));
+    CHECK_OCL_CODE_NO_EXC(clReleaseSampler(m_Sampler));
+  // TODO replace CHECK_OCL_CODE_NO_EXC to CHECK_OCL_CODE and
```

Comment:
Could you provide more info how this change fixes warnings?
Why we need to add to asynchronous exception list? For me it looks like syncronous problem